### ISSUE TITLE
[Fix in Progress] cloudwatch_logs plugin can lose large logs

### ISSUE THREAD

**Issue by PettitWesley** [fluent/fluent-bit#3638]
## Bug Report

**Describe the bug**

I have found that the `cloudwatch_logs` plugin has a bug where it fails to send large logs (> 100 KB) when many are sent in a row. It seems to be some bug in payload creation.

I'm working on a fix, in the meantime you should use the old `cloudwatch` go plugin if: https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit
1. You are currently using `cloudwatch_logs` (it seems likely that many recent Fluent Bit versions are affected by this bug)
2. Your applications emit many large logs over 100KB. I have tested that log loss only occurs with multiple large events sent in series. If you have occasional large events interspersed with smaller logs, its unlikely that you will face any issue.

**Comment by PettitWesley** [comment]
I expect/hope to have a fix up for this within the next 48 hours.

And then hopefully we can do a release @agup006 @edsiper

**Comment by PettitWesley** [comment]
Here are materials to reproduce the issue. I'm planning on eventually storing this and some other stuff in https://github.com/aws/aws-for-fluent-bit to help with testing future reports of log loss.

python script to generate large logs:

```
import datetime
from collections import OrderedDict
import time
import os
import json 
import signal
import random
from fluent import sender

# for local fluent
logger = sender.FluentSender('large-logs-tester')

name = os.environ.get("NAME", "logger")

def print_log(obj):
    if os.environ.get("FLUENT_LOGGER", False):
        logger.emit(name, obj)
        if not logger.emit('large-logs-tester', obj):
            print(logger.last_error)
            logger.clear_last_error() # clear stored error after handled errors
    else:
        print(json.dumps(obj))


PAYLOAD_SIZE = int(os.environ.get("PAYLOAD_SIZE", 32))
PAYLOAD_COUNT = int(os.environ.get("PAYLOAD_COUNT", 100))
MESSAGE_INTERVAL = float(os.environ.get("MESSAGE_INTERVAL", 10))
SIGNAL_PAYLOAD_SIZE = int(os.environ.get("SIGNAL_PAYLOAD_SIZE", 128000))
SIGNAL_PAYLOAD_COUNT = int(os.environ.get("SIGNAL_PAYLOAD_COUNT", 100))

id = 0

def sig_handler(i):
    global id
    for _ in range(10):
        if id < SIGNAL_PAYLOAD_COUNT:
            event_id = int(random.random()*1000)
            print_log(OrderedDict({
                "payload": "x" * SIGNAL_PAYLOAD_SIZE,
                "event_id": event_id,
                "counter": id,
                "global_counter": i, 
                "time": '%s' % datetime.datetime.now(),
            }))
        id += 1
    
def main():
    for i in range(PAYLOAD_COUNT):
        sig_handler(i)
        print("Logged 10 events")
        time.sleep(MESSAGE_INTERVAL)

if __name__ == '__main__': main()
```

Example usage:

```
NAME=run3 FLUENT_LOGGER=1 python3 large.py
```

Fluent Configuration:

```
[INPUT]
    Name forward
    Listen 0.0.0.0
    Port 24224

[SERVICE]
    Parsers_File /home/ec2-user/config/large-logs-testing/parsers.conf
    Flush 1
    Grace 30

[FILTER]
    Name parser
    Match *
    Key_Name log
    Parser json

[OUTPUT]
    Name cloudwatch_logs
    Match   *
    region us-west-2
    log_group_name large-logs-testing
    log_stream_prefix local-repro
    auto_create_group On
```

**Comment by PettitWesley** [comment]
I've put up a fix for the issue. 

Anyone who wants a pre-release image with the fix can use this:

```
144718711470.dkr.ecr.us-west-2.amazonaws.com/issue-3638:latest
```

It can be pulled from any AWS account; this is the repo policy: https://raw.githubusercontent.com/aws/aws-for-fluent-bit/mainline/scripts/public_repo_policy.json

For example:

```
ecs-cli pull --region us-west-2 --registry-id 144718711470 144718711470.dkr.ecr.us-west-2.amazonaws.com/issue-3638:latest
```

The image is built using the AWS for Fluent Bit build process, but it contains the code in my PR against the 1.7 branch. It should go without saying that AWS can not officially provide support for pre-release images.

**Comment by PettitWesley** [comment]
My fix for this has been released in Fluent Bit 1.7.9: https://github.com/aws/aws-for-fluent-bit/releases/tag/v2.16.0
